### PR TITLE
Added test for null values in collections

### PR DESCRIPTION
**Pull request by sagaofsilence**

### CODE REVIEW

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    Function<Collection<?>, Long> countNulls = collection -> collection.stream()
+        .filter(Objects::isNull)
+        .count();
+
+    @Test
+    void givenList_whenNullValueAdded_doesNotFail() {
+        // adding nulls to a list
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        List<Integer> numbers = Arrays.asList(numberArray);
+        assertEquals(countNulls.apply(numbers), 3);
```
Comment (JavierPeris) [inline core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java:33]: Wrong order, please swap these 2 arguments: expected and actual value.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        assertThrows(NullPointerException.class, () -> numbers.get(0)
+            .toString());
+    }
+
+    @Test
+    void givenHashSet_whenNullValueAdded_doesNotFail() {
+        // adding nulls to a HashSet
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        Set<Integer> numbers = new HashSet<>(Arrays.asList(numberArray));
+        assertEquals(countNulls.apply(numbers), 1);
```
Comment (JavierPeris) [inline core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java:48]: Wrong order, please swap these 2 arguments: expected and actual value.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    @Test
+    void givenHashMap_whenNullKeyValueAdded_doesNotFail() {
+        // adding nulls to key value pairs
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        Map<Integer, Integer> numbers = new HashMap<>();
+        Arrays.stream(numberArray)
+            .forEach(integer -> numbers.put(integer, integer));
+        assertEquals(countNulls.apply(numbers.keySet()), 1);
```
Comment (JavierPeris) [inline core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java:71]: Wrong order, please swap these 2 arguments: expected and actual value.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        assertEquals(countNulls.apply(numbers.values()), 1);
```
Comment (JavierPeris) [inline core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java:72]: Wrong order, please swap these 2 arguments: expected and actual value.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        assertThrows(NullPointerException.class, () -> numbers.get(null)
+            .toString());
+    }
+
+    @Test
+    void givenTreeMap_whenNullKeyValueAdded_fails() {
+        // adding nulls to key value pairs
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        Map<Integer, Integer> numbers = new TreeMap<>();
+        assertThrows(NullPointerException.class, () -> Arrays.stream(numberArray)
```
Comment (JavierPeris) [inline core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java:89]: Refactor the code of the lambda to have only one invocation possibly throwing a runtime exception.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java (hunk 1 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java]: Fixed.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java (hunk 2 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java]: Fixed.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java (hunk 3 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java]: Fixed.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java (hunk 4 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java]: Fixed.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java (hunk 5 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java]: We are verifying that a `NullPointerException` is thrown when attempting to insert a `null` key into a TreeMap. I have updated the test. Please review.

File: core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        List<Integer> numbers = Arrays.asList(numberArray);
+        assertEquals(3, countNulls.apply(numbers));
+
+        // accessing nulls from a list
+        assertNull(numbers.get(0));
+
+        // dereferencing nulls from a list
+        assertThrows(NullPointerException.class, () -> numbers.get(0)
+            .toString());
```
Comment (JavierPeris) [inline core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java:41]: What I mean by only one invocation possibly throwing  the exception is we are making two concatenated calls so to make it clearer we can do the following:

```suggestion
        final Integer number = numbers.get(0);
        assertNull(number);
        
        // dereferencing nulls from a list
        assertThrows(NullPointerException.class, () -> number.toString());
```

**Pull request merged**